File: src/main/java/com/kamikazejam/datastore/command/sub/CmdInfo.java
//package com.kamikazejam.datastore.command.sub;
//
//import com.kamikazejam.datastore.DataStoreSource;
//import com.kamikazejam.datastore.util.Color;
//
//public class CmdInfo extends KamiCommand {
//    public CmdInfo() {
//        addAliases("info", "id");
//
//        addRequirements(RequirementHasPerm.get("datastore.command.info"));
//    }
//
//    @Override
//    public void perform() {
//        sender.sendMessage(Color.t("&7--- &6DataStore Information&7---"));
//        sender.sendMessage(Color.t("&7Database Prefix:"));
//        sender.sendMessage(Color.t("  &6" + DataStoreSource.getStoreDbPrefix()));
//        sender.sendMessage(Color.t("&7Storage Service:"));
//        sender.sendMessage(Color.t("  &6Name: " + DataStoreSource.getStorageMode().name()));
//        String r = (DataStoreSource.getStorageService().canCache()) ? "&aYes" : "&cNo";
//        sender.sendMessage(Color.t("  &6Ready: " + r));
//    }
//}

File: src/main/kotlin/com/kamikazejam/datastore/command/sub/CmdInfo.kt
package com.kamikazejam.datastore.command.sub

import com.kamikazejam.datastore.DataStoreSource
import com.kamikazejam.datastore.command.SubCommand
import com.kamikazejam.datastore.util.Color
import org.bukkit.command.CommandSender

class CmdInfo : SubCommand() {
    override val name: String
        get() = "info"

    override val permission: String
        get() = "datastore.command.info"

    override val argsDescription: String
        get() = ""

    override fun execute(sender: CommandSender, args: Array<String>) {
        sender.sendMessage(Color.t("&7--- &6DataStore Information&7---"))
        sender.sendMessage(Color.t("&7Database Prefix:"))
        sender.sendMessage(Color.t("  &6" + DataStoreSource.storeDbPrefix))
        sender.sendMessage(Color.t("&7Storage Service:"))
        sender.sendMessage(Color.t("  &6Name: " + DataStoreSource.storageMode.name))
        val r = if (DataStoreSource.storageService.canCache()) "&aYes" else "&cNo"
        sender.sendMessage(Color.t("  &6Ready: $r"))
    }
}
